client/router_create.go

```diff
@@ -1197,6 +1201,13 @@ sasldb_path: /tmp/qdrouterd.sasldb
 			return err
 		}
 	}
+
+	// Create a generic client secret with the CA associated with the site.
+	err = cli.createGenericClientSecret(siteOwnerRef)
```

Comment:
Is it possible to create the secret by adding the van.TransportCredentials? 

Related suggestion would be to add a 'simple bool' indicator to kube.NewSecret  to consolodate/supplant NewSimpleSecret

Reply:
Done! with this refactor, the secret is created before the router; and this avoids restarting the router after it is created.